packages/block-library/src/post-template/edit.js

```diff
@@ -163,6 +166,54 @@ export default function PostTemplateEdit( {
 			if ( parents?.length ) {
 				query.parent = parents;
 			}
+			if ( postFormat?.length ) {
+				// Get the name and id of the registered post formats from the wp_terms table.
+				const postFormats = getEntityRecords(
+					'taxonomy',
+					'post_format',
+					{
+						order: 'asc',
+						_fields: 'id,name',
+						context: 'view',
+						per_page: -1,
+						post_type: postType,
+					}
+				);
+
+				// Return early if there are no post formats in the database,
+				// and show the "No results found" message.
+				if ( ! postFormats ) {
+					return { posts: [] };
+				}
+
+				// Helper function to compare the value of postFormat to the name in the postFormats object.
+				const isSameTermName = ( termA, termB ) =>
+					termA.toLowerCase() === termB.toLowerCase();
+
+				// Map the postFormat values to the corresponding IDs (term_taxonomy_id).
+				let formatIds = postFormat.map( ( name ) => {
+					const matchingFormat = Object.values( postFormats ).find(
+						( format ) => isSameTermName( format.name, name )
+					);
+					return matchingFormat ? matchingFormat.id : null;
+				} );
+
+				// Convert post format term ID's to a comma separated string,
+				// otherwise the Rest API will return a "rest_invalid_param" error
+				// and there will be a PHP fatal error from _post_format_request.
+				formatIds = formatIds
+					.filter( ( id ) => id !== null )
+					.join( ',' );
+
+				// Return early if there are no matching post formats in the database,
+				// and show the "No results found" message.
+				if ( ! formatIds ) {
+					return { posts: [] };
+				}
+
+				query.post_format = formatIds;
```

Comment:
I'm not sure I understand why we need all the logic here, can't we assign this automatically? Why do we need to do a request to get the right argument?